__react/template/src/channels/$$channel$$.ts.js

```diff
@@ -0,0 +1,83 @@
+import { File } from '@asyncapi/generator-react-sdk';
+import { Publish } from '../../../components/channel/publish';
+import { Subscribe } from '../../../components/channel/subscribe';
+import { Reply } from '../../../components/channel/reply';
+import { Request } from '../../../components/channel/request';
+import { pascalCase, isRequestReply, isReplier, isRequester, isPubsub, messageHasNotNullPayload} from '../../../utils/index';
+
+/**
+ * Return the correct channel component based on whether its `pubSub` or `requestReply`.
+ * 
+ * @param {*} asyncapi 
+ * @param {*} channel to determine the type of
+ * @param {*} channelName 
+ * @param {*} params
+ */
+function getChannelCode(asyncapi, channel, channelName, params) {
+  let channelcode;
+  if (isRequestReply(channel)) {
+    if (isRequester(channel)) {
+      channelcode = Request(
+        asyncapi.defaultContentType(), 
+        channelName, 
+        channel.subscribe().message(0),
+        channel.publish().message(0),
+        channel.parameters()
+      );
+    }
+    if (isReplier(channel)) {
+      channelcode = Reply(
+        asyncapi.defaultContentType(), 
+        channelName, 
+        channel.subscribe().message(0),
+        channel.publish().message(0),
+        channel.parameters(),
+        params
+      );
+    }
+  }
+
+  if (isPubsub(channel)) {
+    if (channel.hasSubscribe()) {
+      channelcode = Publish(
+        asyncapi.defaultContentType(), 
+        channelName, 
+        channel.subscribe().message(0), 
+        channel.parameters());
+    }
+    if (channel.hasPublish()) {
+      channelcode = Subscribe(
+        asyncapi.defaultContentType(), 
+        channelName, 
+        channel.publish().message(0), 
+        channel.parameters());
+    }
+  }
+  return channelcode;
+}
+
+export default function channelRender({ asyncapi, channelName, channel, params }) {
+  // Import the correct messages
+  let publishMessageImport = '';
+  if(channel.hasPublish() && messageHasNotNullPayload(channel.publish().message(0).payload())){
```

Comment:
This same situation with redundant calling some parser functions. `const msg = channel.publish().message(0)` should speed the code :) Here in another places in this function.

Reply:
@magicmatatjahu would you always create a variable for a value if used more then once?

Reply:
In some situation not, but here, calling `channel.publish().message(0)` we first create the `Publish` class's instance and then we create the `Message` class's instance, so I have problem with optimization :)